sections/testingandquality/refactoring.md

```diff
@@ -0,0 +1,27 @@
+# Refactoring
+
+<br/><br/>
+
+### Motivation for Refactoring
+
+Refactoring is an important process in the iterative development flow. Removing "Code Smells" (bad coding practices) such as Duplicated Code, Long Methods, Long Parameter list will improve your code and making it more maintainable. Using a static analysis tools will assist you in finding these code smells and build a process around refactoring.
+<br/><br/>
+
+
+### Martin Fowler - Chief Scientist at ThoughtWorks
+
+ From the book, "Refactoring - Improving the Design of Existing Code"
+
+ > Refactoring is a controlled technique for improving the design of an existing code base.
```

Comment:
@TheHollidayInn Too simplistic, no?

Reply:
It's simple, but a good one, imo. Also Martin Fowler is releasing his second version of this book using Javascript instead of java :P.

I can find another quote if you want.